osc-rest-server/src/main/java/org/osc/core/broker/rest/server/api/VirtualizationConnectorApis.java

```diff
@@ -512,6 +510,53 @@ public Response updateVirtualSecurityPolicyBindings(@Context HttpHeaders headers
     }
 
+    @ApiOperation(value = "Set Security Group Bindings with Service Function Chain  (Openstack Only)",
+            notes = "Adds/Update/Remove Security Group Bindings to Security Function Services.<br/>"
+            		+ "To unbind a service function chain with security group pass null or empty value.<br/>"
+                    + "To Remove all policies, pass in empty json.<br/>"
+                    + "To update services binded to, pass in the updated list of services.<br/>",
+            response = BaseJobResponse.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Successful operation"),
+            @ApiResponse(code = 400, message = "In case of any error", response = ErrorCodeDto.class) })
+    @Path("/{vcId}/securityGroups/{sgId}/sfc/{sfcId}/bindings")
+    @PUT
+    public Response updateVirtualSecurityPolicyBindingsWithSfc(@Context HttpHeaders headers,
+                                                        @ApiParam(value = "The Virtualization Connector Id") @PathParam("vcId") Long vcId,
+                                                        @ApiParam(value = "The Security Group Id") @PathParam("sgId") Long sgId,
+                                                        @ApiParam(value = "Service Function Chain Id, required = false") @PathParam("sfcId") Long sfcId,
+                                                        @ApiParam(value = "List of Policies", required = true) Set<BaseVirtualSystemPoliciesDto> policies) {
+        logger.info("Update Binding SFC and Policies for Security Group - " + sgId);
+        this.userContext.setUser(OscAuthFilter.getUsername(headers));
+        BindSecurityGroupRequest bindRequest = new BindSecurityGroupRequest();
+        bindRequest.setVcId(vcId);
+        bindRequest.setSecurityGroupId(sgId);
+        bindRequest.setSfcId(sfcId);
+        // stub SFC virtual system polices into VirtualSystemPolicyBindingDto
+        for(BaseVirtualSystemPoliciesDto policy : policies) {
+        	VirtualSystemPolicyBindingDto bindDto = new VirtualSystemPolicyBindingDto(policy.getVirtualSystemId(), 
+        																	policy.getName(), policy.getPolicyIds(), policy.getPolicies());
+        	bindRequest.addServiceToBindTo(bindDto);		
+        }
+        return this.apiUtil.getResponseForBaseRequest(this.bindSecurityGroupService, bindRequest);
+    }
+    
+    @ApiOperation(value = "Deletes a Service Function Chain binding with Security Group",
+            notes = "Unbind a Serice Function Chain from a Given Security group and Virtualization Connector")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Successful operation"),
+            @ApiResponse(code = 400, message = "In case of any error", response = ErrorCodeDto.class) })
+    @Path("/{vcId}/securityGroups/{sgId}/sfc")
+    @DELETE
+    public Response unbindSecurityGroupWithServiceFunctionChain(@Context HttpHeaders headers,
```

Comment:
Can you also unbind using the other API? Ideally I think we should have a single API to do that, unless there is a strong reason for having two ways (APIs) to accomplish the same thing.

Reply:
In normal binding scenario , an empty list of binding services unbind Security group binding from binding interfaces. But in SFC case we need to unbind sfc  with security group, so we have pass a null values which is not possible through an url value. hence we opted for a separate API to achieve this functionality.

Reply:
I see but can you use the other API to unbind at all? The documentation has this "To unbind a service function chain with security group pass null or empty value". Either the documentation is not accurate or we need to block unbinding from the other API to allow only a single way to do it.

Reply:
Yes, forgot to remove that line. Updated the notes